akka-http-core/src/main/scala/akka/http/impl/engine/http2/client/ResponseParsing.scala

```diff
@@ -8,15 +8,32 @@ package client
 import akka.annotation.InternalApi
 import akka.http.impl.engine.http2.RequestParsing._
 import akka.http.impl.engine.parsing.HttpHeaderParser
+import akka.http.impl.engine.server.HttpAttributes
+import akka.http.scaladsl.model
 import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers.`Tls-Session-Info`
+import akka.http.scaladsl.settings.ParserSettings
+import akka.stream.Attributes
 import akka.util.OptionVal
 
 import scala.annotation.tailrec
 import scala.collection.immutable.VectorBuilder
 
 @InternalApi
 private[http2] object ResponseParsing {
-  def parseResponse(httpHeaderParser: HttpHeaderParser): Http2SubStream => HttpResponse = { subStream =>
+  def parseResponse(httpHeaderParser: HttpHeaderParser, settings: ParserSettings, attributes: Attributes): Http2SubStream => HttpResponse = { subStream =>
+
+    val tlsSessionInfoHeader: Option[`Tls-Session-Info`] =
+      if (settings.includeTlsSessionInfoHeader) {
+        attributes.get[HttpAttributes.TLSSessionInfo].map(sslSessionInfo =>
+          model.headers.`Tls-Session-Info`(sslSessionInfo.session))
+      } else None
+
+    val tlsSessionAttribute: Option[(AttributeKey[SslSessionInfo], SslSessionInfo)] =
```

Comment:
Yeah, it's a good idea, didn't want to conflict further with Ignasis PR so that's why I didn't DRY it for now, but I see that it is slightly different as well, I'll align.

Reply:
At least aligned now

Reply:
(but let's dry that up at some point when the design has been set whet the other dry-render/parsing-PR)